cryptosolver.py

```diff
@@ -134,11 +133,38 @@ def parse_constraint_input(constraint_string):
 
 # extract the scrambled words from the command-line arguments
 scrambled_words = args
-## HACK: hard code the input for easy testing
-## SENT CENT COST SCENTS SENTIENT SENTENCE
-## UAQJ GAQJ GLUJ UGAQJU UAQJVAQJ UAQJAQGA
+
+## DEBUG HACK: hard code the input for easy testing
+#print("SENT CENT COST SCENTS SENTIENT SENTENCE")
+#print("UAQJ GAQJ GLUJ UGAQJU UAQJVAQJ UAQJAQGA")
 #scrambled_words = "UAQJ GAQJ GLUJ UGAQJU UAQJVAQJ UAQJAQGA".split()
 
+# bail if we have no input
+if len(scrambled_words) == 0:
+    # TODO: print usage info here
+    sys.exit(0)
+
+solution_counter = 0
+scrambled_sentence = " ".join(scrambled_words).upper()
+
+def print_solution(solution):
+    global solution_counter
+    global scrambled_sentence;
+    solution_counter = solution_counter + 1
+    elapsed = round(time.time() - start, 3)
+    print("")
+    print("Solution {}, t={}".format(solution_counter, elapsed))
+    print("")
+    sorted_solution = collections.OrderedDict(sorted(solution.items()))
+    print(" ".join(sorted_solution.keys()))
+    print(" ".join(sorted_solution.values()))
+    # print the scrambled input and the solved output
+    print("")
+    sentence = apply_solution(solution, scrambled_sentence)
+    print(" ", scrambled_sentence)
+    print(" ", sentence)
+    print("")
+
```

Comment:
I prefer to have all the function definitions up near the top rather than blended with most of the main logic below, so I moved `print_solution()` up here.